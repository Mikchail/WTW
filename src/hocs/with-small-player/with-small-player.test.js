import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import withSmallPlayer from './with-small-player';

const noop = () => {};

const MockComponent = (props) => {
  const {onIsPlayingChange, isPlaying} = props;
  return (
    <React.Fragment>
      <div onClick={onIsPlayingChange} isPlaying={isPlaying}>
        Test
      </div>
    </React.Fragment>
  );
};
MockComponent.propTypes = {
  onIsPlayingChange: PropTypes.func,
  isPlaying: PropTypes.bool,
};
const MockComponentWrapped = withSmallPlayer(MockComponent);

it(`withSmallPlayer is rendered correctly`, () => {
  const tree = renderer
    .create(<MockComponentWrapped isPlaying={false} onIsPlayingChange={noop} />, {
      createNodeMock() {
        return {};
      },
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
