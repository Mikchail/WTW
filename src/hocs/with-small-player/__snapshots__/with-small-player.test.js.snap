// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withSmallPlayer is rendered correctly 1`] = `
<div
  isPlaying={false}
  onClick={[Function]}
>
  Test
</div>
`;
